### DISCUSSION TITLE
Middleware option

### DISCUSSION THREAD

**Discussion by devleejb** [vercel/next.js#44633] [Help]
### Describe the feature you'd like to request

Hi,

I'm implementing middleware to block access without authentication.
I wrote middleware.ts at project top level and it works.
However, when the request body size is large, the request doesn't work properly. (The request doesn't reach to pages or api.)
Is there any option for size limit for middleware like in api? (Like below code)

```
export const config = {
    api: {
        bodyParser: {
            sizeLimit: "10mb", // Set desired value here
        },
    },
};
```

Or, is there any other method to implement middleware globally?
I think middleware option is required.

Thank you.

### Describe the solution you'd like

Middleware option like below code

```
export const config = {
    api: {
        bodyParser: {
            sizeLimit: "10mb", // Set desired value here
        },
    },
};
```

### Describe alternatives you've considered

Global option for api (max body size etc...)

**Comment by SamRoehrich** [comment]
Could you provide some more detail or an example repo to show this behavior?
To my knowledge the request body size should not effect the middleware. There could be an issue with your implementation. The docs have an example middleware.js file that shows how to prevent unauthenticated users from hitting your api here. 
https://nextjs.org/docs/advanced-features/middleware#producing-a-response

**Comment by aczekajski** [comment]
It's been two years and there is still no way to set the bodyParser sizeLimit globally, you have to do it individually in each api route handler. And you cannot even import some defaults because the page config has to consist of only static identifiers. This is insane.
